versioned_docs/version-4.0/devupdate.md

```diff
@@ -0,0 +1,958 @@
+  - Initialise the flat navigation by introducing the following in the root layout file (if not existent)
+
+   ```php
+   $nav = $PAGE->flatnav;
+   $templatecontext['firstcollectionlabel'] = $nav->get_collectionlabel();
+   ```
+
+- In order to reintroduce the settings cog in the templates, you can introduce the following:
+
+  ```html
+  <div id="region-main-settings-menu" class="d-print-none {{#hasblocks}}has-blocks{{/hasblocks}}">
+     <div> {{{ output.region_main_settings_menu }}} </div>
+  </div>
+  ```
+
+:::info Accessibility notes
+
+The jump to `maincontent` div is now rendered within the activity header when within an activity context.
+
+:::
+
+## Component library
+
+Each Moodle installation now ships with a Moodle User Interface (UI) Component library, a documentation system used to describe all the Bootstrap components and the custom Moodle components. The component Library is a helper tool for developers when creating user interfaces, a testing tool for theme developers and a documentation tool for core developers. The ultimate goal of having a component library is to encourage developers to create consistent user interfaces to improve Moodle's overall user experience.
+
+The library contains pages with documentation about User Interface components. It contains details on how to use the component, what variations are available and the JavaScript events / options are associated with the component.
+
+When writing on these pages it is possible to render core mustache templates using some custom syntax like this:
+
+```mustache
+ {{< mustache template="core/notification_error" >}}
+ {{< /mustache >}}
+```
+
+You can also call core JavaScript or use HTML examples where the HTML code and the rendered result are visible in the Component Library. For more info visit the [Moodle templates](http://componentlibrary.moodle.com/admin/tool/componentlibrary/docspage.php/library/moodle-templates/) page or the [Moodle JavaScript](http://componentlibrary.moodle.com/admin/tool/componentlibrary/docspage.php/library/moodle-javascript/) page.
+
+Each page in the library uses the current CSS from the default theme in your Moodle installation, if you have multiple themes installed and enabled the setting "Allow theme changes on url", the component library will have a theme selector option.
+
+:::tip
+
+A hosted version of the Component Library can be found here. http://componentlibrary.moodle.com
+
+:::
+
+### Enabling the Component Library
+
+Component library pages are written in the markdown language. These pages need to be compiled to HTML pages before the Component Library is visible. To compile the pages the server running Moodle needs to have [nodeJS and Grunt tools](/general/development/tools/nodejs).
+
+If your server meets all requirements you can enable the library running
+
+```sh
+ $ npm install
+ $ grunt componentlibrary
+```
```

Comment:
```suggestion
```sh
$ npm install
$ grunt componentlibrary
```
```

Reply:
I don't think these blockquotes should be removed.

Reply:
No - that was unintentional. I suspect the suggestion is confused by the amount of backticks